messages.go

```diff
@@ -182,3 +182,126 @@ func NewMsg(msgType string, parent ComposedMsg) (ComposedMsg, error) {
 
 	return msg, nil
 }
+
+// Publish creates a new ComposedMsg and sends it back to the return identities over the
+// IOPub channel.
+func (receipt *msgReceipt) Publish(msgType string, content interface{}) error {
+	msg, err := NewMsg(msgType, receipt.Msg)
+
+	if err != nil {
+		return err
+	}
+
+	msg.Content = content
+	return receipt.SendResponse(receipt.Sockets.IOPubSocket, msg)
+}
+
+// Reply creates a new ComposedMsg and sends it back to the return identities over the
+// Shell channel.
+func (receipt *msgReceipt) Reply(msgType string, content interface{}) error {
+	msg, err := NewMsg(msgType, receipt.Msg)
+
+	if err != nil {
+		return err
+	}
+
+	msg.Content = content
+	return receipt.SendResponse(receipt.Sockets.ShellSocket, msg)
+}
+
+// MIMEDataBundle holds data that can be presented in multiple formats. The keys are MIME types
```

Comment:
Let's move this and any other types up to the top of messages.go, with all the funcs below.